packages/build/src/plugins/spawn.ts

```diff
@@ -35,25 +43,72 @@ const tStartPlugins = async function ({ pluginsOptions, buildDir, childEnv, logs
   logIncompatiblePlugins(logs, pluginsOptions)
 
   const childProcesses = await Promise.all(
-    pluginsOptions.map(({ pluginDir, nodePath, pluginPackageJson }) =>
-      startPlugin({ pluginDir, nodePath, buildDir, childEnv, systemLogFile, pluginPackageJson }),
+    pluginsOptions.map(({ pluginDir, nodePath, nodeVersion, pluginPackageJson }) =>
+      startPlugin({ pluginDir, nodePath, nodeVersion, buildDir, childEnv, systemLogFile, pluginPackageJson }),
     ),
   )
   return { childProcesses }
 }
 
 export const startPlugins = measureDuration(tStartPlugins, 'start_plugins')
 
-const startPlugin = async function ({ pluginDir, nodePath, buildDir, childEnv, systemLogFile, pluginPackageJson }) {
-  const childProcess = execaNode(CHILD_MAIN_FILE, [], {
+const startPlugin = async function ({
+  pluginDir,
+  nodeVersion,
+  nodePath,
+  buildDir,
+  childEnv,
+  systemLogFile,
+  pluginPackageJson,
+}: {
+  nodeVersion: string
+  nodePath: string
+  pluginDir: string
+  /** The process cwd that is used to spawn the child process */
+  buildDir: string
+  childEnv: Record<string, string>
+  pluginPackageJson: Record<string, string>
+  systemLogFile: number
+}) {
+  const ctx = trace.getActiveSpan()?.spanContext()
+
+  // the baggage will be passed to the child process when sending the run event
+  const args = [
+    ...process.argv.filter((arg) => arg.startsWith('--tracing')),
+    `--tracing.traceId=${ctx?.traceId}`,
```

Comment:
`traceId` should already be present in production (pulled from `process.argv`), but we might want to keep this around when running locally I guess 👍

Reply:
yep just for locally